### DISCUSSION TITLE
SWR firing before useContext has loaded

### DISCUSSION THREAD

**Discussion by adherb** [vercel/swr#2002] [Q&A] [answered]
I am making an SWR call to an api and In the query params I am sending a users `user_id` which is coming from useContext.

My issue is the SWR hook is running before context is loading the user's id, meaning that the first SWR call is sending a user Id of undefined, therefore a subsequent SWR call is made. 

What is the best way I can avoid this happening? Is there a way I can allow context to load before the SWR call is made?

**Comment by koba04** [comment, marked as answer]
If you want to wait for sending a request until `user_id` has come, you can use the conditional fetching feature by passing `null` as its key.
https://swr.vercel.app/docs/conditional-fetching

**Reply by adherb** [reply]
Thanks so much!
